src/gui/qgisgui.h

```diff
@@ -83,14 +83,12 @@ namespace QgisGui
   QPair<QString, QString> GUI_EXPORT getSaveAsImageName( QWidget * theParent, QString theMessage, QString defaultFilename = QString::null );
 
   /**
-    Convenience function for readily creating file filters.
-
-    Given a long name for a file filter and a regular expression, return
-    a file filter string suitable for use in a QFileDialog::OpenFiles()
-    call.  The regular express, glob, will have both all lower and upper
-    case versions added.
-  */
-  QString createFileFilter_( QString const &longName, QString const &glob );
+   * Create file filters suitable for use with QFileDialog
+   *
+   * @param format extension e.g. "png"
+   * @return QString e.g. "PNG format (*.png, *.PNG)"
+   */
+  QString createFileFilter_( QString const &format );
```

Comment:
@michaelkirk are you able to merge the upstream code into yours.  As `createFileFilter_` is part of a API you will just have to add two methods for it, e.g add back `QString createFileFilter_( QString const &longName, QString const &glob );`